File: seva-admin/patch.js
const fs = require('fs');
const path = require('path');

const dirPath = __dirname;
const htmlFiles = fs.readdirSync(dirPath).filter(f => f.endsWith('.html'));

const emojiToIcon = {
    '📊': 'dashboard',
    '🧑‍⚕️': 'medical_services',
    '👨‍👩‍👧': 'family_restroom',
    '🧓': 'elderly',
    '📅': 'event',
    '🗓️': 'calendar_month',
    '⏱️': 'schedule',
    '💰': 'payments',
    '💳': 'credit_card',
    '📈': 'bar_chart',
    '🎧': 'headset_mic',
    '⚙️': 'settings',
    '☰': 'menu',
    '🔍': 'search',
    '🔔': 'notifications',
    '👤': 'account_circle',
    '🩺': 'stethoscope'
};

const namesMap = {
    'Sarah Jenkins': 'Sunita Verma',
    'Michael Chen': 'Rahul Desai',
    'Emily Rodriguez': 'Anjali Patel',
    'Robert Johnson': 'Ramesh Kumar'
};

htmlFiles.forEach(file => {
    const filepath = path.join(dirPath, file);
    let content = fs.readFileSync(filepath, 'utf8');

    for (const [oldName, newName] of Object.entries(namesMap)) {
        content = content.split(oldName).join(newName);
    }

    for (const [emoji, icon] of Object.entries(emojiToIcon)) {
        content = content.split(emoji).join(`<span class="material-icons" style="font-size: 1.2rem; transform: translateY(4px);">${icon}</span>`);
    }

    if (!content.includes('Material+Icons')) {
        content = content.replace('</head>', '  <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">\n</head>');
    }

    if (!content.includes('Outfit:wght')) {
        content = content.replace('</head>', '  <link href="https://fonts.googleapis.com/css2?family=Outfit:wght@300;400;500;600;700&display=swap" rel="stylesheet">\n</head>');
    }

    fs.writeFileSync(filepath, content, 'utf8');
});

const cssPath = path.join(dirPath, 'css', 'style.css');
if (fs.existsSync(cssPath)) {
    let css = fs.readFileSync(cssPath, 'utf8');
    css = css.split("'Segoe UI', Tahoma, Geneva, Verdana, sans-serif").join("'Outfit', sans-serif");
    css = css.split("Arial, sans-serif").join("'Outfit', sans-serif");
    fs.writeFileSync(cssPath, css, 'utf8');
}

console.log('Done');

File: seva-admin/patch.py
import os
import re

dir_path = r'c:\Users\ABCD\Desktop\chiu avishkar\seva-admin'
html_files = [f for f in os.listdir(dir_path) if f.endswith('.html')]

# Mapping of emojis to material icons
emoji_to_icon = {
    '📊': 'dashboard',
    '🧑‍⚕️': 'medical_services',
    '👨‍👩‍👧': 'family_restroom',
    '🧓': 'elderly',
    '📅': 'event',
    '🗓️': 'calendar_month',
    '⏱️': 'schedule',
    '💰': 'payments',
    '💳': 'credit_card',
    '📈': 'bar_chart',
    '🎧': 'headset_mic',
    '⚙️': 'settings',
    '☰': 'menu',
    '🔍': 'search',
    '🔔': 'notifications',
    '👤': 'account_circle',
    '🩺': 'stethoscope'
}

# Names replacement
names_map = {
    'Sarah Jenkins': 'Sunita Verma',
    'Michael Chen': 'Rahul Desai',
    'Emily Rodriguez': 'Anjali Patel',
    'Robert Johnson': 'Ramesh Kumar'
}

for file in html_files:
    filepath = os.path.join(dir_path, file)
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Replace names
    for old, new in names_map.items():
        content = content.replace(old, new)
        
    # Replace emojis
    for emoji, icon in emoji_to_icon.items():
        content = content.replace(emoji, f'<span class="material-icons" style="font-size: 1.2rem;">{icon}</span>')
        
    # Add material icons link if not present
    if 'Material+Icons' not in content:
        content = content.replace('</head>', '  <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">\n</head>')
        
    # Add Outfit font link if not present
    if 'Outfit:wght' not in content:
        content = content.replace('</head>', '  <link href="https://fonts.googleapis.com/css2?family=Outfit:wght@300;400;500;600;700&display=swap" rel="stylesheet">\n</head>')
        
    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(content)
        
# Also update style.css to use Outfit
css_path = os.path.join(dir_path, 'css', 'style.css')
if os.path.exists(css_path):
    with open(css_path, 'r', encoding='utf-8') as f:
        css = f.read()
    css = css.replace("font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;", "font-family: 'Outfit', sans-serif;")
    with open(css_path, 'w', encoding='utf-8') as f:
        f.write(css)

print('Done!')
